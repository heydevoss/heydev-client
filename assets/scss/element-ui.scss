// Import Custom Variables
@import "variables";

/* icon font path, required */
$--font-path: '~element-ui/lib/theme-chalk/fonts';

@import "~element-ui/packages/theme-chalk/src/index";