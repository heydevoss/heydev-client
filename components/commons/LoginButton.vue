<template>
  <div>
    <a href="http://localhost:5000/auth/login" class="button--grey">Login</a>
  </div>
</template>

<script>
export default {
  name: 'LoginButton'
}
</script>
<style lang="scss" scoped></style>
