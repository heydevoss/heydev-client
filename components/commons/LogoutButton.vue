<template>
  <div>
    <button class="button--grey" @click="onLogout">Logout</button>
  </div>
</template>

<script>
export default {
  name: 'LogoutButton',
  methods: {
    async onLogout() {
      await this.$apolloHelpers.onLogout()
    }
  }
}
</script>
<style lang="scss" scoped></style>
