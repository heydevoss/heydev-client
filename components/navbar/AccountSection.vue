<template>
  <div>
    <LoginButton v-if="!isAuthenticated" />
    <div v-else class="logged-user">
      <div class="user-infos" @click="redirectToUser">
        <avatar class="avatar" :avatar-url="me.avatarUrl" />
        <span class="user--name">{{ me.name || me.login }}</span>
      </div>
      <LogoutButton class="logout" />
    </div>
  </div>
</template>
<script>
import LoginButton from '@/components/commons/LoginButton'
import LogoutButton from '@/components/commons/LogoutButton'
import Avatar from '@/components/commons/Avatar'
import getUserInfo from '@/apollo/queries/getUserInfo.gql'

export default {
  name: 'AccountSection',
  components: {
    LoginButton,
    LogoutButton,
    Avatar
  },
  computed: {
    isAuthenticated() {
      return !!this.$apolloHelpers.getToken()
    }
  },
  methods: {
    redirectToUser() {
      this.$router.replace(`/user/${this.me.login}`)
    }
  },
  apollo: {
    me: {
      query: getUserInfo
    }
  }
}
</script>
<style lang="scss" scoped>
.user--name:hover {
  cursor: pointer;
}

.avatar {
  width: 40px;
  height: 40px;
  margin-right: 10px;
  cursor: pointer;
}
.logged-user {
  display: flex;
  flex-direction: row;
}
.user-infos {
  display: flex;
  align-items: center;
  margin-right: 5px;
}
.logout {
  margin-left: 10px;
}

@media screen and (max-width: 820px) {
  .user--name {
    visibility: hidden;
  }
  .avatar {
    visibility: hidden;
  }
  .logout {
    margin-left: 0;
  }
}
</style>
