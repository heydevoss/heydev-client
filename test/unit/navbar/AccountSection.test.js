import Vue from 'vue'
import { shallowMount } from '@vue/test-utils'
import AccountSection from '@/components/navbar/AccountSection.vue'
import LogoutButton from '@/components/commons/LogoutButton.vue'
import LoginButton from '@/components/commons/LoginButton.vue'
import { Button } from 'element-ui'

Vue.use(Button)

describe('<AccountSection/>', () => {
  delete window.location
  window.location = { reload: jest.fn() }
  const onLogout = jest.fn()
  const getToken = jest.fn()
  const wrapper = shallowMount(AccountSection, {
    mocks: {
      $apolloHelpers: {
        onLogout,
        getToken
      }
    },
    stubs: {
      'el-button': Button,
      'login-button': LoginButton,
      'logout-button': LogoutButton
    }
  })

  it('shoulbe be a Vue instance and render correctly', () => {
    expect(wrapper.isVueInstance()).toBeTruthy()
    expect(wrapper.element).toMatchSnapshot()
  })
})
