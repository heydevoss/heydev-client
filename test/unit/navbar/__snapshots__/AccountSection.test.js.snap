// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AccountSection/> shoulbe be a Vue instance and render correctly 1`] = `
<div>
  <loginbutton-stub />
</div>
`;
